dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,718 @@
+purpose is to replace GNU/Linux in Google products (like Android, Chrome OS)
+with Zircon microkernel (used to be known as Magenta) and custom userland.
+Fuchsia runs on x86-64 and ARM64. There used to be an
+[unofficial](https://github.com/slavaim/riscv-magenta) RISC-V port, however it
+has been unmaintained for years. Support for 32-bit architectures was dropped
+some time ago.
+
+Zircon is based on LK, however similarities between these two are small and
+constantly decreasing. LK is a kernel intended for embedded usage, while Zircon
+is a general-purpose microkernel that could compete with Linux.
+
+#### Running in DLME
+
+Historically Zircon used Multiboot 1 for booting. Now it uses ZBI with a UEFI
+bootloader called
+[Gigaboot](https://fuchsia.googlesource.com/fuchsia/+/refs/heads/main/src/firmware/gigaboot/).
+Multiboot 1 support is still present, but SKL doesn't support it anyway. See
+[Zephyr](#running-in-dlme) section above for details.
+
+#### Fobnail integration
+
+Fuchsia provides a Unix-like environment with musl C library. Network is
+available but there is no EEM driver, and only XHCI host driver is supported.
+Since EHCI is still widely used, this is a major limitation. TPM driver is
+available but without `libtss2`.
+
+Zircon has a kexec-like capability which allows to boot another Zircon instance.
+This is known as `mexec`.
+
+Following things would have to be done:
+
+* Bring `libtss2` stack to Zircon.
+* Bring EEM driver. There is ECM driver available, which could serve as
+  reference.
+* Bring at least USB EHCI driver, ideally should support UHCI and OHCI too.
+* Figure out how to use `mexec` to boot Linux.
+
+### Xen
+
+Xen is a bare-metal hypervisor available on x86 and ARM. It is able to run Linux
+in a paravirtualized environment. Xen contains only drivers absolutely necessary
+for hypervisor and VM to boot. Control over most of the hardware is handed to
+the first VM known as [Dom0](https://wiki.xenproject.org/wiki/Dom0). Usuallly
+Linux runs in Dom0, there are other OSes which also support running in Dom0
+(BSD, Solaris), however there is no advantage of using them over Linux. Xen also
+has ability to run in dom0-less mode - bootloader provides VM images and Xen
+starts VMs on their own (see section below for details).
+
+#### Running in DLME
+
+Xen is supported by Trenchboot's GRUB and can run without any modifications.
+See [xen-in-dlme.md](xen-in-dlme.md) for demo.
+
+#### Integration with Fobnail
+
+Domain 0 less mode eliminates need for trusted VM. Also, since Xen itself is
+responsible for booting multiple VMs there is no need for tools that manage Xen
+(which don't work on any OS). Fobnail Attester could be integrated in a similar
+way that with CAmkES based VM (see CAmkES section above for details). The main
+difference would be that the secure component would be a separate VM.
+
+Unfortunatelly, Xen is bloated and it doesn't meet the "minimal" requirement.
+This is a major problem.
```

Comment:
So, last resort it makese some sense for early launch, but completly doesn't make sense for future late launch - maybe let's phrase that conclusion correctly.

Reply:
Added [dd5f399](https://github.com/fobnail/docs/pull/26/commits/dd5f399e272e8288493852f605c41b29047c4d16)

Reply:
@TomaszAIR you probably linked different commit, since I see not connection

Reply:
@pietrushnic yes, should be this one [0157aad](https://github.com/fobnail/docs/pull/26/commits/0157aad6fef7b0974587b919f0748211b5e7dc91#diff-76336fb0328e2063bb2c8b183eb9bcf50d8be89da43f108025b2c0624b8f0d6fR522), sorry